haystack_experimental/components/preprocessors/embedding_based_document_splitter.py

```diff
@@ -0,0 +1,402 @@
+            - `documents`: List of documents with the split texts. Each document includes:
+                - A metadata field `source_id` to track the original document.
+                - A metadata field `split_id` to track the split number.
+                - A metadata field `page_number` to track the original page number.
+                - All other metadata copied from the original document.
+
+        :raises:
+            - `RuntimeError`: If the component wasn't warmed up.
+            - `TypeError`: If the input is not a list of Documents.
+            - `ValueError`: If the document content is None or empty.
+        """
+        if not self._is_warmed_up:
+            raise RuntimeError(
+                "The component EmbeddingBasedDocumentSplitter wasn't warmed up. Run 'warm_up()' before calling 'run()'."
+            )
+
+        if not isinstance(documents, list) or (documents and not isinstance(documents[0], Document)):
+            raise TypeError("EmbeddingBasedDocumentSplitter expects a List of Documents as input.")
+
+        split_docs: List[Document] = []
+        for doc in documents:
+            if doc.content is None:
+                raise ValueError(
+                    f"EmbeddingBasedDocumentSplitter only works with text documents but content for "
+                    f"document ID {doc.id} is None."
+                )
+            if doc.content == "":
+                logger.warning("Document ID {doc_id} has an empty content. Skipping this document.", doc_id=doc.id)
+                continue
+
+            doc_splits = self._split_document(doc)
+            split_docs.extend(doc_splits)
+
+        return {"documents": split_docs}
+
+    def _split_document(self, doc: Document) -> List[Document]:
+        """
+        Split a single document based on embedding similarity.
+        """
+        if self.sentence_splitter is None:
+            raise RuntimeError("Sentence splitter is not initialized. Call warm_up() first.")
+        if doc.content is None:
+            raise ValueError("Document content cannot be None.")
+
+        sentences_result = self.sentence_splitter.split_sentences(doc.content)
+        sentences = [sentence["sentence"] for sentence in sentences_result]
+
+        sentence_groups = self._group_sentences(sentences)
+
+        embeddings = self._calculate_embeddings(sentence_groups)
+
+        split_points = self._find_split_points(embeddings)
+
+        splits = self._create_splits_from_points(sentence_groups, split_points)
+
+        # Merge small splits and split large splits
+        final_splits = self._post_process_splits(splits)
+
+        return self._create_documents_from_splits(final_splits, doc)
+
+    def _group_sentences(self, sentences: List[str]) -> List[str]:
+        """
+        Group sentences into groups of sentences_per_group.
+        """
+        if self.sentences_per_group == 1:
+            return [s.strip() for s in sentences]
```

Comment:
Is there a reason we need to run `strip` here?

Reply:
removed